ts/input/tex/TexParser.ts

```diff
@@ -135,10 +141,12 @@ export default class TexParser {
    * @return {ParseResult} The output of the parsing function.
    */
   public parse(kind: HandlerType, input: ParseInput): ParseResult {
-    return this.configuration.handlers.get(kind).parse(input);
+    this.saveI.push(this.i);
+    let result = this.configuration.handlers.get(kind).parse(input);
+    this.updateResult(input[1], this.saveI.pop());
```

Comment:
I think you can do this without needing an array for `saveI`.  Since the value is pushed here and then popped below, with it only be used in one place later (and only to top value is used), you can use a local variable to hold the previous value, save the new value in `saveI` where it can be read later, and then put the local value it back afterward.  That is, you could do

``` js
  public parse(kind: HandlerType, input: ParseInput): ParseResult {
    const i = this.saveI;
    this.saveI = this.i;
    let result = this.configuration.handlers.get(kind).parse(input);
    this.updateResult(input[1], i);
    this.saveI = i;
    return result;
  }
```

The make the change to `Push()` listed below.